packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -306,8 +309,16 @@ class SliverChildBuilderDelegate extends SliverChildDelegate {
   /// The [builder], [addAutomaticKeepAlives], [addRepaintBoundaries],
   /// [addSemanticIndexes], and [semanticIndexCallback] arguments must not be
   /// null.
+  ///
+  /// Providing a [findItemIndex] lets [SliverChildBuilderDelegate]
```

Comment:
Suggestion:

If the order in which [builder] returns children ever changes, consider providing a [findItemIndex]. This allows the delegate to find the new index for a child that was previously located at a different index to attach the existing state to the Widget at its new location.